sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1724,23 +1637,25 @@ public void validate(PCollection<TableRow> input) {
 
       // Exactly one of the table and table reference can be configured.
       checkState(
-          jsonTableRef != null || tableRefFunction != null,
+          getJsonTableRef() != null || getTableRefFunction() != null,
           "must set the table reference of a BigQueryIO.Write transform");
       checkState(
-          jsonTableRef == null || tableRefFunction == null,
+          getJsonTableRef() == null || getTableRefFunction() == null,
           "Cannot set both a table reference and a table function for a BigQueryIO.Write"
               + " transform");
 
       // Require a schema if creating one or more tables.
       checkArgument(
-          createDisposition != CreateDisposition.CREATE_IF_NEEDED || jsonSchema != null,
+          getCreateDisposition() != CreateDisposition.CREATE_IF_NEEDED || getJsonSchema() != null,
           "CreateDisposition is CREATE_IF_NEEDED, however no schema was provided.");
 
       // The user specified a table.
-      if (jsonTableRef != null && validate) {
+      BigQueryServices bqServices =
+          MoreObjects.firstNonNull(getBigQueryServices(), new BigQueryServicesImpl());
```

Comment:
if getBigQueryService() returns non null, we still create a throwaway object here, correct?